src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -42,6 +42,10 @@ public ODataMessageReaderSettings()
             Validator = new ReaderValidator(this);
         }
 
+        /// <summary>Gets or sets the OData protocol version to be used for reading payloads. </summary>
+        /// <returns>The OData protocol version to be used for reading payloads.</returns>
+        public ODataVersion? Version { get; set; }

```

Comment:
I was thinking about adding this in addition to the existing maxVersion. Wondering why OData didn't have such a need in its long history?

Reply:
Technically, you only need to specify the Version of the payload you're reading if reading is ambiguous between formats. We typically try to make sure that things we add/change in the format are unambiguous, favoring adding something new over changing the meaning of something that exists.  Even for reading deltas, we don't really need to know the version for the reader, but it seemed a good thing to add in general.  Writing, of course, does need this because the writer generates different output when writing different formats.

---
In reply to: [150159325](https://github.com/OData/odata.net/pull/989#discussion_r150159325) [](ancestors = 150159325)